qa/os/src/test/java/org/elasticsearch/packaging/util/Docker.java

```diff
@@ -291,6 +291,22 @@ public static void copyFromContainer(Path from, Path to) {
 
             return super.getScriptCommand("docker exec --user elasticsearch:root --tty " + containerId + " " + script);
         }
+
+        @Override
+        public Result run(String script) {
+            try {
+                return super.run(script);
+            } catch (ShellException e) {
+                try {
+                    final Shell.Result dockerLogs = getContainerLogs();
+                    throw new ShellException(
+                        e.getMessage() + "\n\nContainer stdout: " + dockerLogs.stdout + "\n\nContainer stderr: " + dockerLogs.stderr
+                    );
+                } catch (ShellException logsException) {
+                    throw new ShellException("Command failed, and couldn't get docker logs", logsException);
```

Comment:
We lose the original exception in this scenario and effectively replace it with one about not being able to fetch logs. We should probably still rethrow the original exception and maybe add this one as a suppressed exception.